python_analytics/event_encoder.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import absolute_import, unicode_literals
+
+from abc import ABCMeta, abstractmethod
+from weakref import WeakKeyDictionary
+
+from six import add_metaclass, text_type
+
+
+NoValue = object()
+
+
+@add_metaclass(ABCMeta)
+class _TrackedAttribute(object):
```

Comment:
While, it would be nice, I am not sure that it is essential for this PR. Do you need to be more specific in the size of string or int?